apps/builder/app/builder/features/workspace/use-read-canvas-rect.ts

```diff
@@ -16,12 +16,25 @@ export const useReadCanvasRect = () => {
     null
   );
 
+  // react updates dom asynchronously so here schedule rect
+  // computing into effect when react done with dom
+  const [updateCallback, setUpdateCallback] = useState<
+    undefined | (() => void)
+  >(undefined);
+  useEffect(() => {
+    updateCallback?.();
+  }, [updateCallback]);
```

Comment:
why not?
```suggestion
  useEffect(updateCallback, [updateCallback]);
```

Reply:
because effect does not accept undefined